traversal/predicate/Predicate.java

```diff
@@ -39,13 +41,59 @@ private Predicate(PRED_OP operator, PRED_ARG argument) {
         this.hash = Objects.hash(operator, argument);
     }
 
+    // TODO: Move this where it belongs - Encoding?
+    public static int compareBooleans(boolean first, boolean second) {
```

Comment:
Comparison methods for values of compatible data-types. Considered the source-of-truth